### PR TITLE
Static analyzer fixes

### PR DESCRIPTION
**Pull request by jpakkane**
I ran the code through the analyzer at lgtm.com. This MR fixes the low hanging fruit.

### CODE REVIEW

**Comment by ignatenkobrain** [comment]
@jpakkane btw, should we enable lgtm as our CI as well?

**Comment by jpakkane** [comment]
That is something we could do, yes. But if you look at [the result page](https://lgtm.com/projects/g/mesonbuild/meson/alerts/) there are a bunch of warnings (mostly due to multiple inheritance in compilers) that would probably need to be fixed first. Hitting those would be confusing for newcomers.

File: tools/ac_converter.py [hunk 1]
```diff
@@ -39,7 +39,6 @@
      'HAVE_POSIX_MEMALIGN': ('posix_memalign', 'stdlib.h'),
      'HAVE_SIGACTION': ('sigaction', 'signal.h'),
      'HAVE_ALARM': ('alarm', 'unistd.h'),
-     'HAVE_CLOCK_GETTIME': ('clock_gettime', 'time.h'),
```
Comment (nirbheek) [inline tools/ac_converter.py]: Why remove this?

Comment (jpakkane) [inline tools/ac_converter.py]: Because it was in the dictionary twice with identical values.

Comment (nirbheek) [inline tools/ac_converter.py]: Aah, LGTM then ;)

**Pull request merged**